src/interactions/slashCommands/SlashCommandBuilder.ts

```diff
@@ -0,0 +1,122 @@
+import type { APIApplicationCommandOption } from 'discord-api-types/v8';
+import { mix } from 'ts-mixer';
+import { validateMaxOptionsLength, validateRequiredParameters } from './Assertions';
+import { SharedNameAndDescription, SharedSlashCommandOptions } from './SlashCommandOptions';
+import { SlashCommandSubCommandBuilder, SlashCommandSubCommandGroupBuilder } from './SlashCommandSubCommands';
+
+@mix(SharedSlashCommandOptions, SharedNameAndDescription)
+export class SlashCommandBuilder {
+	protected name: string = undefined!;
+
+	protected description: string = undefined!;
+	protected options: ToAPIApplicationCommandOptions[] = [];
+
+	/**
+	 * Returns the final data that should be sent to Discord. You won't need this unless you're manually
+	 * creating slash commands via this builder.
+	 *
+	 * **Note:** Calling this getter will validate required properties based on their conditions.
```

Comment:
There's `@note` too, although we can use `<warn></warn>` in this organization.

Reply:
We shouldn't rely too much on docs comments too much yet honestly.

Reply:
> There's `@note` too, although we can use `<warn></warn>` in this organization.

warn tags don't render in any IDE that I know off either. I believe it's fine as is

Reply:
@note doesnt render either?

Reply:
They *sort of do*, but I still wouldn't use them. Whats wrong with the current approach?

Reply:
Nothing, just asking.